### PR TITLE
"istioctl describe" Log to scope; custom warnings for missing Istio rules

### PR DESCRIPTION
**Pull request by esnible**
Resolves https://github.com/istio/istio/issues/21815

This PR does three things
1. Pilot's configClient log messages are placed under a scope
2. Istioctl suppresses warnings from that scope
3. `istioctl x describe` has new warnings for stale config_dump that refers to non-existent Istio rules.

### CODE REVIEW

File: pilot/pkg/config/kube/crd/controller/client.go [hunk 1]
```diff
@@ -84,6 +84,8 @@ type restClientBuilder struct {
 	types          []*crd.SchemaType
 }
 
+var scope = log.RegisterScope("client", "", 0)
```
Comment (ayj) [inline pilot/pkg/config/kube/crd/controller/client.go]: `kube` maybe? client is pretty generic and not what I would expect to enable for kube logs.

Comment (esnible) [reply pilot/pkg/config/kube/crd/controller/client.go]: Now using 'kube'.

**Comment by howardjohn** [comment]
/retest

There was a broken test that is fixed

**Pull request merged**